milli/src/search/new/matches/mod.rs

```diff
@@ -821,30 +880,32 @@ mod tests {
     fn format_highlight_crop_phrase_query() {
         //! testing: https://github.com/meilisearch/meilisearch/issues/3975
         let temp_index = TempIndex::new();
+
+        let text = "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!";
         temp_index
             .add_documents(documents!([
-                { "id": 1, "text": "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!" }
+                { "id": 1, "text": text }
             ]))
             .unwrap();
+
         let rtxn = temp_index.read_txn().unwrap();
 
         let format_options = FormatOptions { highlight: true, crop: Some(10) };
-        let text = "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!";
 
         let builder = MatcherBuilder::new_test(&rtxn, &temp_index, "\"the world\"");
         let mut matcher = builder.build(text, None);
         // should return 10 words with a marker at the start as well the end, and the highlighted matches.
         insta::assert_snapshot!(
             matcher.format(format_options),
-            @"…had the power to split <em>the</em> <em>world</em> between those who…"
+            @"…had the power to split <em>the world</em> between those who…"
```

Comment:
Well, it's weird, but their are still 10 words and the match is centered in the crop window so I feel this behavior is better 🤔

The question is, why the previous behavior was wrong in some ways, maybe the center of the window were computed from the first match 🤔

Reply:
The crop algorithm start from the match, then, using a bidirectional iterator, adds a word switching side growing the window until the word count is reached, like:
```
         <em>the world</em>
   SPLIT <em>the world</em>
   split <em>the world</em> BETWEEN
TO split <em>the world</em> between
to split <em>the world</em> between THOSE
```

unless we encounter an hard-separator like `.` or `;` in this case we will prefer add a word in the same "sentence" to have a contextualized crop window.

Reply:
> The question is, why the previous behavior was wrong in some ways, maybe the center of the window were computed from the first match 🤔

Yeah, I think as I detailed [here](https://github.com/meilisearch/meilisearch/pull/4928#issuecomment-2350899477), somewhat poorly worded, it's because it counts from the first match of a phrase, while my solution counts from the beginning and the end of the phrase the positions. I'm not entirely sure yet that everything will work as it should, I might want to add more tests, but for now I'm pushing the changes with the tests adjusted.